node/src/options.rs

```diff
@@ -13,34 +13,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Node configuration options
+//! The node command line options.
 
-use std::ffi::OsString;
-use std::net::SocketAddr;
-use std::path::PathBuf;
+use std::{ffi::OsString, net::SocketAddr, path::PathBuf};
+
+use clap::{Args, Parser, Subcommand};
 use strum::VariantNames;
 
 use common::chain::config::ChainType;
 
 /// Mintlayer node executable
-#[derive(clap::Parser, Debug)]
+#[derive(Parser, Debug)]
 #[clap(author, version, about)]
 pub struct Options {
     /// Where to write logs
     #[clap(long, value_name = "PATH")]
     pub log_path: Option<PathBuf>,
 
-    /// Address to bind RPC to
-    #[clap(long, value_name = "ADDR", default_value = "127.0.0.1:3030")]
-    pub rpc_addr: SocketAddr,
+    #[clap(subcommand)]
+    pub command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Command {
+    /// Create a configuration file.
+    CreateConfig {
+        /// The path where config will be created.
+        // TODO: Use a system-specific location by default such as `%APPDATA%` on Windows and
+        // `~/Library/Application Support` on Mac.
+        #[clap(short, long, default_value = "./.mintlayer/mintlayer.toml")]
+        path: PathBuf,
+    },
+    Run(RunOptions),
+}
+
+#[derive(Args, Debug)]
+pub struct RunOptions {
+    /// The path to the configuration file.
+    // TODO: Use a system-specific location by default such as `%APPDATA%` on Windows and
+    // `~/Library/Application Support` on Mac.
+    #[clap(short, long, default_value = "./.mintlayer/mintlayer.toml")]
+    pub config_path: PathBuf,
 
-    /// Blockchain type
+    /// Blockchain type.
     #[clap(long, possible_values = ChainType::VARIANTS, default_value = "mainnet")]
     pub net: ChainType,
 
-    /// Address to bind P2P to
-    #[clap(long, value_name = "ADDR", default_value = "/ip6/::1/tcp/3031")]
-    pub p2p_addr: String,
+    /// Blockchain type.
+    #[clap(long)]
+    pub max_db_commit_attempts: Option<usize>,
+
+    /// The maximum capacity of the orphan blocks pool.
+    #[clap(long)]
+    pub max_orphan_blocks: Option<usize>,
```

Comment:
The comment will show up in the help message. It should clarify the capacity is specified in the number of blocks. People may think it's a limit on the number of bytes or something like that and open themselves up to a DoS attack.

Reply:
I agree that a documentation, especially that is directly user-facing should be as clear as possible.

And thank you for pointing out the typos too!